### PR TITLE
Quest Student Library: Refactor

### PR DESCRIPTION
**Pull request by fobadara**
The following requirements were met:
- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsibility Principle".
    - Make sure that the role of `main.rb` is only to start your code and ask the user for the initial choice of options (or to quit).
    - Extract some pieces of logic into a separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -19,22 +12,17 @@
 # app = App.new
 # app.run
```
Comment (andersonlebon) [inline app.rb]: - [ ] Kindly remove commented code to clean this file.

File: app.rb [hunk 2]
```diff
@@ -56,17 +44,23 @@ def menu
...
-      create_rental
+      Creator.create('Rental')
+      menu
     when 6
-      list_rentals_by_person_id
+      Listing.list('Rentals')
+      menu
     when 7
       puts `clear`
       puts 'Thank you for using Quest Student Library. See you next time!'
```
Comment (andersonlebon) [inline app.rb]: - [ ] Kindly make sure the `main.rb` is one that handles options (ask the user for the initial choice of options (or to quit).) As the requirements says :point_down:

#### Make sure that the role of` main.rb` is only to start their code and ask the user for the initial choice of options (or to quit).

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @team

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (fobadara) [reply app.rb]: Thank you

File: app.rb (hunk 2 above)
Comment (fobadara) [reply app.rb]: We have implemented the requested changes.

**Comment by Amrhub** [comment]
# Invalid 🔴 

Hi @fobadara , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 

You have submitted a project, that has linters failing

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by EricMbouwe — APPROVED** [review]
### Hi @team, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tresorsawasawa** [comment]
Hey @EricMbouwe 

Thank You for the review :pray: 
We really appreciate :+1: :+1:

**Pull request merged**